components/docs-chef-io/content/habitat/origin-keys.md

```diff
@@ -0,0 +1,190 @@
+
+![Example form content for uploading an origin key in Builder](/images/screenshots/builder-key-upload.png)
+
+## Managing Origin Keys with the CLI
+
+Run Chef Habitat CLI commands from your local environment or from within the Chef Habitat Studio.
+
+See the CLI documentation for more information on the [`hab origin key`]({{< relref "habitat-cli.md/#hab-origin-key" >}}) commands.
+
+### Find Your Origin Keys
+
+Chef Habitat stores your public and private origin keys at `~/.hab/cache/keys` on Linux systems, `C:\hab\cache\keys` on Windows, and at `/hab/cache/keys` inside of the Chef Habitat Studio environment.
+
+#### Find Origin Keys in a Local Environment
+
+On Windows:
+
+```PowerShell
+Get-ChildItem C:\hab\cache\keys
+```
+
+On Linux or macOS:
+
+```bash
+ls -la ~/.hab/cache/keys
+```
+
+#### Find Origin Keys in the Chef Habitat Studio
+
+On Windows:
+
+```powershell
+Get-ChildItem C:\hab\cache\keys
+```
+
+On Linux or macOS:
+
+```bash
+ls -la /hab/cache/keys
+```
+
+### Generate Origin Keys
+
+When you create an origin through the site, Chef Habitat Builder automatically generates an origin key pair.
+
+The Chef Habitat CLI creates origin key pairs through two different commands, for two different uses:
+
+- Use [`hab setup`]({{< relref "install-habitat.md" >}}) to generate your first origin key pair as part of setting up the `hab` CLI
+- Use the `hab origin key generate <ORIGIN>` command to create an key pair for an origin created with the `hab origin create` command
+
+Create origin keys with the `hab` command:
+
+```hab
+hab origin key generate <ORIGIN>
+```
+
+### Download Origin Keys
+
+To get your public origin key using the command line, use:
+
+```hab
+hab origin key download <ORIGIN>
+```
+
+### Upload Origin Keys
+
+Creating an origin with the `hab origin create` command registers the origin on Chef Habitat Builder without creating an origin key pair. The `hab origin key generate` command creates the key pair and saves them in your local environment, but it does not upload either origin key to Chef Habitat Builder.
+
+- Only "administrators" and "owners" can upload new keys to an origin.
+- Builder requires the public origin key to upload artifacts for that origin, so you'll need to upload it.
+- Builder requires the private origin key to enable new artifact builds from packages with plans linked to that origin.
+
+Upload origin keys with the `hab` command:
+
+```hab
+hab origin key upload <ORIGIN>
+```
+
+Upload the origin private key:
+
+```hab
+hab origin key upload --secret <ORIGIN>
+```
+
+Upload both origin keys at the same time:
+
+```hab
+hab origin key upload  --secfile <PATH_TO_PRIVATE_KEY> --pubfile <PATH_TO_PUBLIC_KEY>
+```
+
+### Import Origin Keys
+
+Use `hab origin key import` to read the key from a standard input stream into Chef Habitat Builder:
+
+```hab
+hab origin key import <enter or paste key>
+hab origin key import <PATH_TO_KEY>
+curl <URL_THAT_RETURNS_KEY> | hab origin key import
```

Comment:
Just to keep things simple (and more true-to-life), I'd probably replace this `curl` with something like 

```bash
cat <PATH_TO_KEY> | hab origin key import
```

Reply:
ok